### PR TITLE
minimal bottomsheet example

### PR DESCRIPTION
**Pull request by toretefre**
Trying to close https://github.com/AtB-AS/kundevendt/issues/8886.

Just a minimal example of what I'm trying to do now, not ready for merge.

### CODE REVIEW

File: src/components/sections/items/date-input/DateInputSectionItem.ios.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function DateInputSectionItem(props: DateInputSectionItemProps) {
         display="compact"
         testID="dateInput"
         minimumDate={new Date()}
-        maximumDate={maximumDate}
```
Comment (toretefre) [inline src/components/sections/items/date-input/DateInputSectionItem.ios.tsx:42]: This is set to mimic Android behaviour on iOS.

File: src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/StartTimeSelection.tsx [hunk 2]
```diff
@@ -89,6 +100,16 @@ export function StartTimeSelection({
           },
         ]}
       />
+      {showActivationDateWarning && (
```
Comment (toretefre) [inline src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/StartTimeSelection.tsx:103]: This one **is** shown immediately on date change

File: src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravelDate/TravelDateSheet.tsx [hunk 3]
```diff
@@ -83,11 +116,21 @@ export const TravelDateSheet = forwardRef<ScrollView, Props>(
           <Section>
             <DateInputSectionItem
               value={dateString}
-              onChange={setDate}
+              onChange={onSetDate}
               maximumDate={maximumDate}
             />
             <TimeInputSectionItem value={timeString} onChange={setTime} />
           </Section>
+          {showActivationDateWarning && (
```
Comment (toretefre) [inline src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravelDate/TravelDateSheet.tsx:124]: This one is **not** showing immediately on date change, only when re-mounting the bottom sheet.

**Comment by toretefre** [comment]
Hey @gorandalum (or code owner @marius-at-atb), 

Do you have any idea why state changes in parent root stack does not propagate in realtime to the `TravelDateSheet`? 

In this example, the yellow warning box in the `TravelDateSheet` should appear at the same time as the one in the background (`Root_PurchaseOverviewScreen.tsx`), but it does not appear until I re-open the `TravelDateSheet`. 

My hypothesis is that the `TravelDateSheet` in some way is disconnected from state updates until re-opening, but I can't see why.

See video and notice the yellow `MessageBox` appearing immediately behind the bottom sheet.

https://github.com/AtB-AS/mittatb-app/assets/21310942/69b1ecad-fd62-4478-90dc-7ad8f5196805

Let me know if I can clarify something. If you want to test with AtB tickets you could do so by setting `latestActivationDate` in limitations of a product in staging as done for [fram-staging Vaksen Buss](https://github.com/AtB-AS/firestore-configuration/blob/d7de8eae9f61d3bf91abdec1ddf92fa46504dbdd/documents/fram-staging/referenceData.json#L785C31-L785C31).

**Comment by marius-at-atb** [comment]
@toretefre I think the props for TravelDateSheet are set when calling openTravelDateSheet. TravelDateSheet then updates its internal state while "alive", but openTravelDateSheet is not called reactively when the props have changed.
Not sure if we have an existing way to handle this @gorandalum?

If there is no good way to update the props from outside, I guess it is possible to create an internal state variable that mirrors showActivationDateWarning inside TravelDateSheet 😬

**Comment by toretefre** [comment]
Good idea @marius-at-atb, I was thinking about something similar, but decided it might not be best practice.

There is a PR for this solution here, if possible it would be great to have a review before the weekend so we could go ahead with FRAM release: https://github.com/AtB-AS/mittatb-app/pull/3889

**Comment by toretefre** [comment]
Closed in favour of https://github.com/AtB-AS/mittatb-app/pull/3889, thanks for good help!

**Comment by gorandalum** [comment]
> @toretefre I think the props for TravelDateSheet are set when calling openTravelDateSheet. TravelDateSheet then updates its internal state while "alive", but openTravelDateSheet is not called reactively when the props have changed. Not sure if we have an existing way to handle this @gorandalum?
> 
> If there is no good way to update the props from outside, I guess it is possible to create an internal state variable that mirrors showActivationDateWarning inside TravelDateSheet 😬

In general we update the local state internally in the bottom sheet continuously, and update the external state when pressing save. This is the same as done in the `TravelSearchFiltersBottomSheet`.

**Pull request closed without merging**